index.js

```diff
@@ -121,22 +124,21 @@ Liftoff.prototype.buildEnvironment = function (opts) {
     }, this);
   }
 
-  // use rechoir to autoload any required modules
-  var autoloads;
-  if (configPath) {
-    autoloads = rechoir.prepare(this.extensions, configPath, cwd, true);
-    if (autoloads instanceof Error) {
-      autoloads = autoloads.failures;
-    }
-    if (Array.isArray(autoloads)) {
-      autoloads.forEach(function (attempt) {
-        if (attempt.error) {
-          this.emit('requireFail', attempt.moduleName, attempt.error);
-        } else {
-          this.emit('require', attempt.moduleName, attempt.module);
+  registerLoader(this.extensions, configPath, cwd, this);
```

Comment:
I'd recommend moving the `this` from the last argument to the first.  Maybe this function signature is getting too complex and should be changed to an object with each argument as a property.  @tkellen thoughts?